core/storage/trie/impl/trie_storage_impl.cpp

```diff
@@ -21,7 +21,7 @@ namespace kagome::storage::trie {
       boost::optional<std::shared_ptr<changes_trie::ChangesTracker>> changes) {
     // will never be used, so content of the callback doesn't matter
     auto empty_trie = trie_factory->createEmpty(
-        [](const auto &branch, auto idx) { return nullptr; });
+        [](auto &branch) { return outcome::success(); });
```

Comment:
```suggestion
        [](auto &node) { return outcome::success(); });
```

Reply:
If you look at the code in PR, you'll see how to retrieveNode is more opaque before calling any specific node methods. And less error prone also.

Reply:
You may mention that I call retrieveNode just 3 times in PR, just once in every function: detachNode, deleteNode, handleDeletion. What it would be like when you use retrieveChild? You need to call it before any children request. That would be way more than 3. Or you need to be sure where to call it and decrease the number of calls. Anyway it is harder. What's the point of opaqueness if it complexes the code. When trie logic complete and code coveraged it won't change much. As you may mention it wasn't touch for quite long before clearPrefix task.